src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -202,11 +211,29 @@ public DiagnosticResult EmitAssembly(string outputPath)
 
                 Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
 
+                var diagnostics = new List<Diagnostic>(emitResult.Diagnostics);
+
+                if (CompilationContext.DiagnosticAnalyzers != null && CompilationContext.DiagnosticAnalyzers.Length > 0)
+                {
+                    sw.Start();
+
+                    var analyzerDiagnostics = CompilationContext.Compilation
+                        .WithAnalyzers(CompilationContext.DiagnosticAnalyzers)
+                        .GetAllDiagnosticsAsync()
+                        .Result     // TODO: make the caller an async method?
```

Comment:
File this issue: https://github.com/aspnet/dnx/issues/2386

I think we should keep the interface as-is for now.